pulp_rpm/app/models/repository.py

```diff
@@ -321,26 +324,34 @@ class RpmDistribution(PublicationDistribution):
     def content_handler(self, path):
         """Serve config.repo and public.key."""
         if path == self.repository_config_file_name:
-            val = f"""[{self.name}]
-enabled=1
-baseurl={settings.CONTENT_ORIGIN}{settings.CONTENT_PATH_PREFIX}{self.base_path}/
-gpgcheck=0
-"""
+            base_url = f"{settings.CONTENT_ORIGIN}{settings.CONTENT_PATH_PREFIX}{self.base_path}/"
+            publication = self.publication.cast()
+            val = textwrap.dedent(
+                f"""\
+                [{self.name}]
+                enabled=1
+                baseurl={base_url}
+                gpgcheck={publication.gpgcheck}
+                """
+            )
             repository_pk = self.publication.repository.pk
             repository = RpmRepository.objects.get(pk=repository_pk)
             signing_service = repository.metadata_signing_service
             if signing_service is None:
```

Comment:
i believe we should not look at the presence of the signing service any more but instead take as a reference values for gogcheck set in the  publication

Reply:
@dkliban can you confirm my understanding?

Reply:
If a signing service exists, we will still need to look for the path of the public key. Therefore, In my opinion, it is necessary to have such an if-condition there.

Also, in the current state, users can still specify whether they want to perform GPG signature checks even though they published a repository with a signing service.

Reply:
I actually don't know if it is a valid config when you have gpgcheck set to 1 but gpgkey would not be provided so I guess let's keep the check for the signing service

Reply:
Not sure i understand all of the discussion here, i guess atm only the repository has a reference to the signing service.
But in the future, we want the signing service to be immutable and contain the public key, so it can be referenced by the also immutable publication.
Not sure, what's the best solution today.

Reply:
FWIW, the pulp_deb plugin provides the signing_service only with the publication.